src/Boot/BootstrapHook.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Drush\Boot;
+
+use Consolidation\AnnotatedCommand\Hooks\InitializeHookInterface;
+use Symfony\Component\Console\Input\InputInterface;
+use Consolidation\AnnotatedCommand\AnnotationData;
+
+/**
+ * The BootstrapHook is installed as an init hook that runs before
+ * all commands. If there is a `@bootstrap` annotation, then we will
+ * bootstrap Drupal to the requested phase.
+ */
+class BootstrapHook implements InitializeHookInterface
+{
+    protected $bootstrapManager;
+
+    public function __construct(BootstrapManager $bootstrapManager)
+    {
+        $this->bootstrapManager = $bootstrapManager;
+    }
+
+    public function initialize(InputInterface $input, AnnotationData $annotationData)
+    {
+        // Check the @bootstrap annotation. If there isn't one, then exit.
+        if (!$annotationData->has('bootstrap')) {
+            return;
+        }
+
+        $phase = $annotationData->get('bootstrap');
+        if (!defined($phase)) {
+            return;
```

Comment:
We should log a warning or something here.

It would also probably be nice to allow folks to use shorter aliases, e.g. `full` instead of `DRUSH_BOOTSTRAP_FULL`, via a lookup table.